publication-doi-commons/src/main/java/no/unit/nva/publication/doi/PublicationMapper.java

```diff
@@ -0,0 +1,159 @@
+import no.unit.nva.publication.doi.dto.Publication;
+import no.unit.nva.publication.doi.dto.Publication.Builder;
+import no.unit.nva.publication.doi.dto.PublicationDate;
+import no.unit.nva.publication.doi.dto.PublicationType;
+import nva.commons.utils.JsonUtils;
+
+public class PublicationMapper {
+
+    public static final String DEFAULT_ROOT = "/dynamodb";
+    public static final String PUBLICATION_TYPE = "Publication";
+    public static final String ERROR_MUST_BE_PUBLICATION_TYPE = "Must be a dynamodb stream record of type Publication";
+    protected static final ObjectMapper objectMapper = JsonUtils.objectMapper;
+
+    private static final JsonPointer CONTRIBUTOR_ARP_ID_JSON_POINTER = JsonPointer.compile("/m/identity/m/arpId/s");
+    private static final JsonPointer CONTRIBUTOR_NAME_JSON_POINTER = JsonPointer.compile("/m/identity/m/name/s");
+
+    private static final JsonPointer CONTRIBUTORS_LIST_POINTER = JsonPointer.compile(
+        "/newImage/entityDescription/m/contributors/l");
+
+    private static final JsonPointer PUBLICATION_IDENTIFIER_POINTER
+        = JsonPointer.compile("/newImage/identifier/s");
+    private static final JsonPointer PUBLICATION_TYPE_POINTER = JsonPointer.compile(
+        "/newImage/entityDescription/m/reference/m/publicationInstance/m/type/s");
+    private static final JsonPointer PUBLICATION_ENTITY_DESCRIPTION_POINTER = JsonPointer.compile(
+        "/newImage/entityDescription/m");
+    private static final JsonPointer DOI_POINTER = JsonPointer.compile(
+        "/newImage/entityDescription/m/reference/m/doi/s");
+    private static final JsonPointer MAIN_TITLE_POINTER = JsonPointer.compile(
+        "/newImage/entityDescription/m/mainTitle/s");
+    private static final JsonPointer TYPE_POINTER = JsonPointer.compile("/newImage/type/s");
+    private static final JsonPointer INSTITUTION_OWNER_POINTER = JsonPointer.compile("/newImage/publisherId/s");
+
+    private final String root;
+
+    public PublicationMapper() {
+        this(DEFAULT_ROOT);
+    }
+
+    /**
+     * Construct a PublicationMapper where json pointer for data lookups is prefixed with ROOT. (because a Publication
+     * DTO payload can be wrapped under other json structures)
+     *
+     * @param root root for Publication Dynamodb Event or Stream Record.
+     */
+    public PublicationMapper(String root) {
+        this.root = root;
+    }
+
+    /**
+     * Map to doi.Publication from a dynamo db stream record from nva_publication / nva_resources
+     *
+     * @param publicationIdPrefix                        prefix for a publication, from running environment
+     *                                                   (https://nva.unit.no/publication)
+     * @param json detail.dynamodb serialized as a string
+     * @return Publication doi.Publication
+     * @throws IOException on IO exception
+     */
+    public Publication fromDynamodbStreamRecord(String publicationIdPrefix,
```

Comment:
Maybe move this to the constructor?

This needs to be populated by the running `Environment`.